src/otx/core/data/transform_libs/torchvision.py

```diff
@@ -6,17 +6,129 @@
 from __future__ import annotations
 
 from inspect import isclass
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any, Sequence
 
 import hydra
+import torch
 import torchvision.transforms.v2 as tvt_v2
+from torch import Tensor
+from torchvision.transforms.v2 import functional as F  # noqa: N812
+from torchvision.tv_tensors import BoundingBoxes, Image
 
 if TYPE_CHECKING:
     from torchvision.transforms.v2 import Compose
 
     from otx.core.config.data import SubsetConfig
 
 
+class PerturbBoundingBoxes(tvt_v2.Transform):
+    """Perturb bounding boxes with random offset values."""
+
+    def __init__(self, offset: int) -> None:
+        super().__init__()
+        self.offset = offset
+
+    def _transform(self, inpt: Any, params: dict[str, Any]) -> Any:  # noqa: ANN401
+        if isinstance(inpt, BoundingBoxes):
+            mean = torch.zeros_like(inpt)
+            repeated_size = torch.tensor(inpt.canvas_size).repeat(len(inpt), 2)
+            std = torch.minimum(repeated_size * 0.1, torch.tensor(self.offset))
+            noise = torch.normal(mean, std)
+            return BoundingBoxes(
+                (inpt + noise).clamp(mean, repeated_size - 1),
+                format=inpt.format,
+                canvas_size=inpt.canvas_size,
+                dtype=inpt.dtype,
+            )
+        return inpt
```

Comment:
This is somewhat far away from how we define `tvt_v2.Transform`. You need to implement `kernel` and register `tv_tensors` to the `kernel`. Can we address this in the next PR?

Reply:
Sure. I didn't fully understand how to set custom v2 transform.
So, I'll try to change it following v2 transform flow while this PR is open.

Reply:
I tried it like below, but it seems that we cannot register custom transform to kernel because `F.register_kernel` only supports torchvision functional.

```python
def perturb_bounding_boxes(inpt: torch.Tensor, offset: int) -> torch.Tensor:
    """Perturb bounding boxes with offset value."""
    mean = torch.zeros_like(inpt)
    repeated_size = torch.tensor(inpt.canvas_size).repeat(len(inpt), 2)
    std = torch.minimum(repeated_size * 0.1, torch.tensor(offset))
    noise = torch.normal(mean, std)
    return (inpt + noise).clamp(mean, repeated_size - 1)

@F.register_kernel(perturb_bounding_boxes, tv_tensors.BoundingBoxes)
def _perturb_bounding_boxes_dispatch(inpt: tv_tensors.BoundingBoxes, offset: int) -> tv_tensors.BoundingBoxes:
    output = perturb_bounding_boxes(inpt.as_subclass(torch.Tensor), offset)
    return tv_tensors.wrap(output, like=inpt)
```

Reply:
Would you escalate this issue about adding custom kernel to the upstream? https://github.com/pytorch/vision/issues

Reply:
https://github.com/pytorch/vision/issues/8223